src/synapse/azext_synapse/_params.py

```diff
@@ -43,12 +39,20 @@ def load_arguments(self, _):
                        help='The size of the returned list.By default it is 20 and that is the maximum.')
 
     with self.argument_context('synapse spark batch create') as c:
-        c.argument('language', arg_type=get_enum_type(['SCALA', 'PYTHON', 'DOTNET'], default='SCALA'),
+        c.argument('main_definition_file', help='The main file used for the job.')
+        c.argument('main_class_name',
+                   help='The fully-qualified identifier or the main class that is in the main definition file.')
+        c.argument('command_line_arguments', nargs='+',
+                   help='Optional arguments to the job (Note: please use storage URIs for file arguments).')
+        c.argument('archives', nargs='+', help='The array of archives.')
+
+    with self.argument_context('synapse spark batch create') as c:
+        c.argument('language', arg_type=get_enum_type(SparkBatchLanguage, default=SparkBatchLanguage.Scala),
                    help='The spark batch job language.')
 
     for scope in ['show', 'cancel']:
         with self.argument_context('synapse spark batch ' + scope) as c:
-            c.argument('batch_id', options_list=['--id'], arg_group='Spark Batch',
+            c.argument('batch_id', options_list=['--id', '--livy-id'], arg_group='Spark Batch',

```

Comment:
>'--id',  [](start = 49, length = 8)

We might need to avoid such `--id` parameter name according to Azure CLI guidelines since it can be a common alias. [https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#write-a-command](https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#write-a-command) . You decide it.

Reply:
removed `--id`

---
In reply to: [418080571](https://github.com/Azure/azure-cli-extensions/pull/1581#discussion_r418080571) [](ancestors = 418080571)